src/metadata/ForeignKeyMetadata.ts

```diff
@@ -87,8 +88,14 @@ export class ForeignKeyMetadata {
         this.referencedEntityMetadata = options.referencedEntityMetadata;
         this.columns = options.columns;
         this.referencedColumns = options.referencedColumns;
-        this.onDelete = options.onDelete || "NO ACTION";
-        this.onUpdate = options.onUpdate || "NO ACTION";
+    
+        /**
+         * "For an ON DELETE or ON UPDATE that is not specified, the default action is always RESTRICT."
+         * https://dev.mysql.com/doc/refman/8.0/en/create-table-foreign-keys.html
+         */
+        this.onDelete = options.onDelete || (this.entityMetadata.connection.driver instanceof MysqlDriver) ? "RESTRICT" : "NO ACTION";
```

Comment:
Not sure why its throwing [this error](https://travis-ci.org/typeorm/typeorm/jobs/523736835#L1665)

This condition seems to be failing for some reason. Any suggestions? 
```typescript 
(this.entityMetadata.connection.driver instanceof MysqlDriver) 
```